packages/@aws-cdk/aws-eks-next/lib/cluster.ts

```diff
@@ -922,6 +905,15 @@ export class Cluster extends ClusterBase {
     this.clusterSecurityGroupId = resource.attrClusterSecurityGroupId;
     this.clusterEncryptionConfigKeyArn = resource.attrEncryptionConfigKeyArn;
 
+    this.kubectlSecurityGroup = ec2.SecurityGroup.fromSecurityGroupId(this, 'ClusterSecurityGroup', this.clusterSecurityGroupId);
+
+    this.connections = new ec2.Connections({
+      securityGroups: [this.kubectlSecurityGroup],
```

Comment:
It's a bit confusing that you are reusing `kubctlSecurityGroup` here. This should be `this.clusterSecurityGroup` (they could both be the same value but just from a readability perspective it's not clear)